File: Python/Medium/arrays_hashing/convert1Dto2DArray.py
import collections
def solution(nums):
    if len((set(nums))) == len(nums):
        return [nums]
    
    counter = collections.Counter(nums)
    max_val = max(counter.values())
    res = [[] for _ in range(max_val)]

    i = 0
    while i<max_val+1:
        for k,v in counter.items():
            if counter[k] == 0: continue
            res[v-1].append(k)
            counter[k] -= 1
        i += 1
    
    return res

File: JavaScript/Medium/arrays_hashing/convert1Dto2DArray.js
var solution = function(nums){
    let temp = new Set(nums)
    if (temp.size == nums.length){
        return [nums]
    }

    let counter = {}
    for (let i = 0; i< nums.length; i++){
        if (!counter[nums[i]]) { 
            counter[nums[i]] = 1
        } else {
            counter[nums[i]] ++
        }
    }

    let max_val = Math.max(...Object.values(counter))
    let res = new Array(max_val).fill(0).map(() => new Array())

    let i = 0
    while (i < max_val){
        for (const [k,v] of Object.entries(counter)){
            if (counter[k] === 0) continue
            res[v-1].push(k)
            counter[k] -- 
        }
        i ++
    }

    return res
}